Libraries/Image/Image.ios.js

```diff
@@ -143,6 +139,14 @@ var Image = React.createClass({
     var isStored = !source.isStatic && !isNetwork;
     var RawImage = isNetwork ? RCTNetworkImage : RCTStaticImage;
 
+    var {width, height} = source;
+    var style = flattenStyle([{width, height}, styles.base, this.props.style]);
+    if (isNetwork) {
+      // NetworkImageView uses a measure method instead to set original width and height
+      style = flattenStyle([styles.base, this.props.style]);
```

Comment:
you're throwing away the work done in flattenStyle above - it can be quite expensive, so only do one or the other.